### ISSUE TITLE
Feature: Revoke Tokens

### ISSUE THREAD

**Issue by GoldenGnu** [burberius/eve-esi#46]
It’s now possible to revoke access and refresh tokens.
We should add support for that.
- https://eveonline-third-party-documentation.readthedocs.io/en/latest/sso/tokenrevocation.html
- https://github.com/ccpgames/sso-issues/issues/7

**Comment by deadlybulb** [comment]
If we don't want to do much work, we can wedge this in using the existing ApiClient with something like this:

```java
String clientID = ...; // These are EVE app parameters
String secretKey = ...;
String encoded = "Basic " + Base64.encodeToString((clientID + ":" + secretKey).getBytes("UTF-8"), false)
String tokenType = ...; // should be "access_token" or "refresh_token"
String token = ...; // token to revoke
ApiClient invoker = new ApiClient();
invoker.setBasePath("https://login.eveonline.com");
try {
  // "better" way is to use a collector into a map, but omitted for clarity 
  Map<String, String> headers = new HashMap<>();
  headers.put("Authorization", encoded);
  headers.put("Content-Type", "application/x-www-form-urlencoded");
  headers.put("Host", "login.eveonline.com");
  Map<String, Object> form = new HashMap<>();
  form.put("token_type_hint", tokenType);
  form.put("token", token);
  invoker.invokeAPI("/oath/revoke", "POST", null, "", headers, form, null,
                                 "application/x-www-form-urlencoded", new String[0], null); 
  // revocation succeeded!
} catch (ApiException e) {
  // revocation failed
}
```

An alternative is to use the `javax.ws.rs.client` libraries directly with the implementation provided by  Jersey.  That's a lower level interface but has the virtue of being independent of the swagger codegen generated ApiClient.

Either way, we also need to choose a package location.  I was thinking `net.troja.eve.esi.util` to avoid colliding with the auto-gen code.

Thoughts?

**Comment by GoldenGnu** [comment]
Thank you for taking a look at it 👍 

I was thinking we should add it to the `net.troja.eve.esi.api.SsoApi` class

Something like:
```java
    private static final String ACCESS_TOKEN = "access_token";
    private static final String REFRESH_TOKEN = "refresh_token";
    private static final String URI_REVOKE = "https://login.eveonline.com";

    public void revokeRefreshToken(String refreshToken) throws ApiException {
        revokeToken(refreshToken, REFRESH_TOKEN);
    }

    public void revokeAccessToken(String refreshToken) throws ApiException {
        revokeToken(refreshToken, ACCESS_TOKEN);
    }
    
    private void revokeToken(String token, String tokenTypeHint) throws ApiException {
        final String basePath = apiClient.getBasePath(); // Save old basePath
        apiClient.setBasePath(URI_REVOKE);
        ...
        try {
            apiClient.invokeAPI(...);
        } finally {
            apiClient.setBasePath(basePath); // load old basePath
        }
    }
```

I tried your code and it's very close to what I had myself, but, neither are working...
I'm just getting 401

**Comment by GoldenGnu** [comment]
Fixed in #48

**Comment by burberius** [comment]
Just a test for GoldenGnu
